### ISSUE TITLE
Support numbered list for flowchart direction?

### ISSUE THREAD

**Issue by sheeeng** [mermaid-js/mermaid#6111]
### Description

```mermaid
flowchart LR
    Start -- "1. stop" --> Stop
    Stop -- "2. start" --> Start
```

### Steps to reproduce

```mermaid
flowchart LR
    Start -- "1. stop" --> Stop
    Stop -- "2. start" --> Start
```

### Screenshots

![image](https://github.com/user-attachments/assets/0a324c25-e687-48ed-b578-92f760ba73a9)

### Code Sample

```text
flowchart LR
    Start -- "1. stop" --> Stop
    Stop -- "2. start" --> Start
```

**Comment by ryan-brandt** [comment]
I don't think this is an error/bug. If you escape the markdown, then you can have a list. For instance the following code with escaped markdown renders like this:
```
flowchart LR
    Start -- "1\. stop" --> Stop
    Stop -- "2\. start" --> Start
```

```mermaid
flowchart LR
    Start -- "1\. stop" --> Stop
    Stop -- "2\. start" --> Start
```

Because it is markdown, if the change was made to allow a markdown list, then it would act like a markdown list. This means markdown would determine the number for each item in the list. So this markdown would render like:
```
flowchart LR
    Start -- "1. stop" --> Stop
    Stop -- "2. start" --> Start
```

<img width="369" alt="Image" src="https://github.com/user-attachments/assets/202f0f5c-7d32-44a5-ad30-3d75590fb3e9" />

Note how even though the numbers were `1.` and `2.` markdown updated all of the values to be `1.` since it considers each number to be a new list.

If anything, this seems like to would be a new feature to allow for lists to be dynamically managed across arrows.

**Comment by knsv** [comment]
[sisyphus-bot]

Thanks for the question! This is actually expected behavior related to how Markdown parsing works. The text `1. stop` is being interpreted as a Markdown ordered list item before Mermaid processes it.

You can work around this by escaping the period:

```
flowchart TD
    1\. stop --> 2\. move
```

Or by wrapping the node text in quotes:

```
flowchart TD
    A["1. stop"] --> B["2. move"]
```

The second approach gives you the most control over how the text is displayed.

Thanks for raising this — it's a common gotcha, and your question will help others who run into the same thing!
